packages/react-notifications/src/components/notifications-list/notifications-list.tsx

```diff
@@ -0,0 +1,273 @@
+import React from 'react';
+import { css } from '@emotion/core';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  NOTIFICATION_DOMAINS,
+  NOTIFICATION_KINDS_GLOBAL,
+  NOTIFICATION_KINDS_PAGE,
+  NOTIFICATION_KINDS_SIDE,
+  TAppNotificationDomain,
+  TAppNotificationGlobal,
+  TAppNotificationPage,
+  TAppNotificationSide,
+  TAppNotificationValuesUnexpectedError,
+} from '@commercetools-frontend/constants';
+import { removeNotification } from '@commercetools-frontend/notifications';
+import { customProperties } from '@commercetools-frontend/ui-kit';
+import Notification from '../notification';
+import GenericNotification from '../notification-kinds/generic';
+import ApiErrorNotification from '../notification-kinds/api-error';
+import UnexpectedErrorNotification from '../notification-kinds/unexpected-error';
+import { useCustomNotificationComponent } from '../map-notification-to-component';
+import {
+  selectGlobalNotifications,
+  selectPageNotifications,
+  selectSideNotifications,
+} from './selectors';
+import { TAppState } from './types';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+
+type Props = {
+  domain: TAppNotificationDomain;
+};
+
+const getStyles = (props: Props) => {
+  const baseStyles = css`
+    color: ${customProperties.colorSurface};
+    position: relative;
+    width: 100%;
+    z-index: 10000;
+  `;
+
+  switch (props.domain) {
+    case NOTIFICATION_DOMAINS.GLOBAL:
+      return css`
+        ${baseStyles};
+        text-align: center;
+        width: 100% !important;
+
+        > {
+          display: inline-flex;
+          padding-right: 48px;
+          height: 100%;
+          width: 100%;
+        }
+      `;
+    case NOTIFICATION_DOMAINS.PAGE:
+      return css`
+        ${baseStyles};
+        padding: ${customProperties.spacingM} ${customProperties.spacingL};
+      `;
+    case NOTIFICATION_DOMAINS.SIDE:
+      return css`
+        ${baseStyles};
+        position: absolute;
+        text-align: left;
+        height: 0;
+        overflow: visible;
+      `;
+    default:
+      return css``;
+  }
+};
+
+const NotificationsList = (props: Props) => {
+  const dispatch = useDispatch();
+  const mapCustomNotificationToComponent = useCustomNotificationComponent();
+
+  switch (props.domain) {
+    case NOTIFICATION_DOMAINS.GLOBAL: {
+      const notifications = useSelector<TAppState, TAppNotificationGlobal[]>(
+        selectGlobalNotifications
+      );
+      return (
+        <div css={getStyles(props)}>
+          {notifications.map(notification => {
+            // 1. Check if there is a custom notification component first
+            const CustomNotificationComponent = mapCustomNotificationToComponent(
+              notification
+            ) as Notification;
```

Comment:
Since we don't know the custom notification type, we need to "cast" it to `Notification` in order to pass the `notification` prop and adhere with its types.